packages/wrangler/templates/startDevWorker/ProxyWorker.ts

```diff
@@ -305,10 +305,12 @@ function insertLiveReloadScript(
  */
 function rewriteUrlRelatedHeaders(headers: Headers, from: URL, to: URL) {
 	headers.forEach((value, key) => {
-		if (typeof value === "string" && value.includes(from.host)) {
+		if (typeof value === "string" && value.includes(from.hostname)) {
 			headers.set(
 				key,
-				value.replaceAll(from.origin, to.origin).replaceAll(from.host, to.host)
+				value
+					.replaceAll(from.origin, to.origin)
+					.replaceAll(from.hostname, to.hostname)
```

Comment:
I think we also want to replace the host as well, in case the port has changed...

```suggestion
					.replaceAll(from.origin, to.origin)
					.replaceAll(from.host, to.host)
					.replaceAll(from.hostname, to.hostname)
```

Reply:
I am a little worried though that we would, for example, now replace all `localhost` references in all headers whether or not they are actually referencing the correct host (e.g. you might have a header that refers to some other port on localhost and this would get swapped.

I wonder if we need to special case the "cookie" header in this case?

Reply:
I think we might need to, yes. Browsers don't seem to accept `Set-Cookie` on localhost with a port specified, and in any case cookies are shared across ports in general.